packages/features/portfolio/src/containers/portfolio-state-zero/hooks/use-token-data.tsx

```diff
@@ -0,0 +1,98 @@
+import {
+  useAllNetworkMarkets,
+  useAllUniqueUnderlyingTokens,
+  useSelectedNetwork,
+} from '@notional-finance/notionable-hooks';
+import { PRODUCTS } from '@notional-finance/util';
+
+export const useTokenData = (selectedTabIndex: number, activeToken: string) => {
+  const selectedNetwork = useSelectedNetwork();
+  const depositTokens = useAllUniqueUnderlyingTokens([selectedNetwork]);
+  const { earnYields, borrowYields } = useAllNetworkMarkets();
+
+  const requiredProducts = {
+    0: ['Variable Lend', 'Fixed Lend', 'Provide Liquidity'],
+    1: ['Leveraged Liquidity', 'Leveraged Vault'],
+    2: ['Variable Borrow', 'Fixed Borrow'],
+  };
+
+  const yieldData = selectedTabIndex === 2 ? borrowYields : earnYields;
+
+  const tokenObj: {
+    [symbol: string]: { products: string[]; data: typeof earnYields };
+  } = {};
+
+  const leveragedTokenData: Record<string, any> = {};
+
+  if (selectedTabIndex === 1) {
```

Comment:
Can you create named constants for the different tabs so these are easier to read?

Reply:
the mui tabs element needs numbers. I can create a object with names at the component level if you think that would help

Reply:
Yes thats what I mean

Reply:
Cool, lmk if you're happy with my solution